pkg/operator/controller/ingressclass/ingressclass.go

```diff
@@ -0,0 +1,139 @@
+package ingressclass
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/google/go-cmp/cmp/cmpopts"
+
+	operatorv1 "github.com/openshift/api/operator/v1"
+	routev1 "github.com/openshift/api/route/v1"
+	"github.com/openshift/cluster-ingress-operator/pkg/operator/controller"
+
+	networkingv1 "k8s.io/api/networking/v1"
+
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// ensureIngressClass ensures an IngressClass exists for the IngressController
+// with the given name.  Returns a Boolean indicating whether the IngressClass
+// exists, the current IngressClass if it does exist, and an error value.
+func (r *reconciler) ensureIngressClass(ingresscontrollerName string, ingressclasses []networkingv1.IngressClass) (bool, *networkingv1.IngressClass, error) {
+	want, desired := desiredIngressClass(ingresscontrollerName, ingressclasses)
+
+	have, current, err := r.currentIngressClass(ingresscontrollerName)
+	if err != nil {
+		return false, nil, err
+	}
+
+	switch {
+	case !want && !have:
+		return false, nil, nil
+	case !want && have:
+		if err := r.client.Delete(context.TODO(), current); err != nil {
+			if !errors.IsNotFound(err) {
+				return true, current, fmt.Errorf("failed to delete IngressClass: %v", err)
```

Comment:
Use `%w` instead of `%v` here, and in the other `fmt.Errorf` calls in this file, if you don't mind :grin: